APMrover2/RC_Channel.cpp

```diff
@@ -0,0 +1,266 @@
+    AP_SUBGROUPINFO(obj_channels[1], "2_",  2, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 3_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[2], "3_",  3, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 4_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[3], "4_",  4, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 5_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[4], "5_",  5, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 6_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[5], "6_",  6, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 7_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[6], "7_",  7, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 8_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[7], "8_",  8, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 9_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[8], "9_",  9, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 10_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[9], "10_", 10, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 11_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[10], "11_", 11, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 12_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[11], "12_", 12, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 13_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[12], "13_", 13, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 14_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[13], "14_", 14, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 15_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[14], "15_", 15, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 16_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[15], "16_", 16, RC_Channels_Rover, RC_Channel_Rover),
+
+    AP_GROUPEND
+};
+
+
+Mode *Rover::mode_from_mode_num(const enum Mode::number num)
+{
+    Mode *ret = nullptr;
+    switch (num) {
+    case Mode::number::MANUAL:
+        ret = &mode_manual;
+        break;
+    case Mode::number::ACRO:
+        ret = &mode_acro;
+        break;
+    case Mode::number::STEERING:
+        ret = &mode_steering;
+        break;
+    case Mode::number::HOLD:
+        ret = &mode_hold;
+        break;
+    case Mode::number::AUTO:
+        ret = &mode_auto;
+        break;
+    case Mode::number::RTL:
+        ret = &mode_rtl;
+        break;
+    case Mode::number::SMART_RTL:
+        ret = &mode_smartrtl;
+        break;
+    case Mode::number::GUIDED:
+       ret = &mode_guided;
+        break;
+    case Mode::number::INITIALISING:
+        ret = &mode_initializing;
+        break;
+    default:
+        break;
+    }
+    return ret;
+}
+
+int8_t RC_Channels_Rover::flight_mode_channel_number() const
```

Comment:
Why not move the flight mode channel into the RC_Channels object? Other then the param conversion code (which is a one liner in this case), it removes a bunch of helpers from each vehicle's version of the object.